### ISSUE TITLE
Concurrency issue when clicking on chart with high frequency data updates

### ISSUE THREAD

**Issue by EricZeiberg** [beto-rodriguez/LiveCharts2#1081]
**Describe the bug**
I have a chart which receives new data points at around 2KHz (2000/s) on a separate thread and plots them. 

While performance is mostly fine, I noticed that when clicking on the chart, it will (after a couple clicks), produce a `System.InvalidOperationException (Collection Was Modified)` exception and crash. This is most likely due to one thread adding and removing data points from the `ObservableCollection<ObservablePoint>` array and the UI thread trying to read from the same collection. 

I'm using a thread lock and the `SyncContext` property to prevent this but the click handler code doesn't seem to utilize this. I have tooltips turned off as well. 

**To Reproduce**
1. Create a chart and add points to it at a high frequency on a separate thread (non-UI thread)
2. Click a couple times anywhere on the chart window
3. Observe the library crash

**Expected behavior**
The chart should not crash. 

**Desktop:**
 - OS: Windows 10

**Additional context**
After debugging, it seems like the problematic code is here, in [`Kernel/Extensions.cs`](https://github.com/beto-rodriguez/LiveCharts2/blob/60958aacd622793df898b341f8098a49009217ab/src/LiveChartsCore/Kernel/Extensions.cs#L481). 
```c#
public static ChartPoint? FindClosestTo(this IEnumerable<ChartPoint> points, LvcPoint point)
    {
        var fp = new LvcPoint((float)point.X, (float)point.Y);

        return points
            .Select(p => new
            {
                distance = p.DistanceTo(fp),
                point = p
            })
            .OrderBy(p => p.distance)
            .FirstOrDefault()?.point;
    }
```
This LINQ expression does not lock the `SyncContext` object before it performs a `Select()` query.

**Comment by beto-rodriguez** [comment]
Hi and thanks for the report.

When your data updates from multiple threads you must either invoke the changes in the UI thread or provide a sync context for the chart to update.

Please see [this issue](https://github.com/beto-rodriguez/LiveCharts2/issues/225#issuecomment-1012488255) for more info. I will close this for now since the library already provides a solution for this scenario, let me know if you face any issue implementing the solution for this.

**Comment by beto-rodriguez** [comment]
I am not able to reproduce this, it probably means that the concurrency issue could be on your side, please could you provide a repository with the issue?

You can test the library by cloning this repo and go to the multithreading sample, then to create a similar case to yours just modify the `ReadData` method to:

```c#
private async void ReadData()
    {
        await Task.Delay(1000);

        while (true)
        {
            await Task.Delay(_delay);

            for (var i = 0; i < 2000; i++)
            {
                _current = Interlocked.Add(ref _current, _r.Next(-9, 10));
                lock (Sync)
                {
                    _values.Add(_current);
                    _values.RemoveAt(0);
                }
                await Task.Delay(1);
            }
        }
    }
```

Finally set these 2 variables in the constructor of the VM:

```c#
_delay = 1000;
var readTasks = 5;
```

With those settings we are adding/removing about 10,000 points per seconds I event can use tooltips without issues:

![multithreading](https://github.com/beto-rodriguez/LiveCharts2/assets/10853349/78de69e0-2c0c-4ad2-87f6-18f0197c459b)

I will close this for now since there is no evidence that the library has an issue here, please feel free to reply or open a new issue with a repository where I can see the issue and help you better.

**Comment by beto-rodriguez** [comment]
Which platform is it?

**Comment by EricZeiberg** [comment]
Avalonia 11. Was also breaking with the 0.10.x version. Sorry, forgot to mention this.

**Comment by EricZeiberg** [comment]
Thanks for investigating. 

Yes, I can invoke in UI thread but posting on that thread 2000 times / sec causes performance issues on my end.

**Comment by beto-rodriguez** [comment]
You were right, when you click the chart, it tries to invoke the chart events, but since the data is changing on another thread this exception is thrown, it is strange that it does not throw in the rest on the platforms 😥 

The referenced commit fixes this, this is just for now, I think we can do much better and completely ignore the click event when there are no handlers subscribed to the chart events.

**Comment by EricZeiberg** [comment]
Thanks for the quick fix! When should I expect a build with this change to be published to NuGet?

**Comment by beto-rodriguez** [comment]
It is included now in [2.0.0-beta.850.1](https://www.nuget.org/packages/LiveChartsCore/2.0.0-beta.850.1)
